### PR TITLE
Test: section testing

### LINKED ISSUE #1151: Create src/components/sections/bounties component test

### PR DESCRIPTION
**Pull request by Samantha-KY**
# Submit a pull request

- [ ] This is not a duplicate of an existing pull request.
- [ ] No existing features have been broken without good reason.
- [ ] Your commit messages are detailed
- [ ] The code style guideline have been followed.
- [ ] Documentation has been updated to reflect your changes.
- [ ] Tests have been added or updated to reflect your changes.
- [ ] All tests pass.

---

Replace any ":question:" below with information about your pull request.

## Pull Request Details

Provide details about your pull request and what it adds, fixes, or changes.
Closes #1151 

:question:

## Breaking Changes

Describe what features are broken by this pull request and why, if any.

:question:

## Issues Fixed

Enter the issue numbers resolved by this pull request below, if any.

1. :question:

## Other Relevant Information

Provide any other important details below.

## Summary by CodeRabbit

- **New Features**
  - Introduced mock data for components such as challenges, communities, courses, referrals, rewards, scoreboard, and submissions to enhance testing and simulation.

- **Bug Fixes**
  - Improved property access in `CourseCard` component to prevent potential `null` or `undefined` errors.
  - Added `data-testid` attributes for better testability in components (`MainHeader`, `Sidebar`, `Wrapper`).

- **Tests**
  - Added new test cases for components: `BountiesNavigation`, `ChallengeHeader`, `Learning`, `Objectives`, `Rewards`, `Rubric`, `SetupTeamChallenge`, `Submission`, `CoursesOverview`, `MainHeader`, `Sidebar`, `Wrapper`, and `ScoreboardOverview`.

### CODE REVIEW

File: __mocks__/submission.ts
Comment (Jonath-z) [inline __mocks__/submission.ts:1]: this file needs to be deleted

File: __mocks__/user.ts
Comment (Jonath-z) [inline __mocks__/user.ts:1]: this one as well

File: __mocks__/challenge.ts [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+}
+
+export const mockTeam: Team = {
+  challenge: challenge(),
+  challenge_ref: "challenge ref",
+  created_at: "created at",
+  id: "id",
+  locked: true,
+  name: "Master",
+  organizer: mockUser(),
```
Comment (Jonath-z) [inline __mocks__/challenge.ts:143]: why is mock user a function? I think we should keep the code consistent, let's avoid mocking with functions

File: __tests__/components/sections/bounties/Navigation.test.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  });
+
+  it("should render menu items", () => {
+    renderWithRedux(<BountiesNavigation />);
+    const menuItems = screen.getAllByRole("listitem");
+    menuItems.forEach(() => {
+      expectedNavItems.forEach((menu) => {
+        if (!menu.hideTitle) {
+          const menuTitle = menu.title;
+          expect(screen.getByText(menuTitle)).toBeInTheDocument();
```
Comment (Jonath-z) [inline __tests__/components/sections/bounties/Navigation.test.tsx:40]: Shouldn't only check the element to be in the document you can also check for in the title is the one you mocked

File: __tests__/components/sections/bounties/Navigation.test.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        }
+      });
+    });
+  });
+
+  it("renders links with href", () => {
+    renderWithRedux(<BountiesNavigation />);
+    const links = screen.getAllByRole("link");
+    links.forEach((link) => {
+      expect(link).toHaveAttribute("href");
```
Comment (Jonath-z) [inline __tests__/components/sections/bounties/Navigation.test.tsx:50]: Can you also check the value of the href?

**Review by Jonath-z — COMMENTED** [review]
I left some comment which are repetitive, checking if an element is in the DOM is not enough for test, we need to check if the data or mock we have passed is the one present in the DOM, (the more our tests resemble the way our components are used the more confidence they give us) 

- https://medium.com/@babux1/the-core-principles-of-testing-in-react-with-react-testing-library-9aa708889291#:~:text=Principle%201%3A%20Test%20User%20Behavior,users%20interact%20with%20your%20components.

File: __mocks__/challenge.ts (hunk 1 above)
Comment (Samantha-KY) [reply __mocks__/challenge.ts:143]: This is the fix for the other error we were having of calling the mock before initialization

File: __tests__/components/sections/communities/overview/scoreboard/index.test.tsx
Comment (serapieTuyishime) [inline __tests__/components/sections/communities/overview/scoreboard/index.test.tsx:1]: could you also test for when the button is clicked to reveal more content!

File: src/components/sections/challenges/Learning.tsx [hunk 4]
```diff
@@ -24,10 +24,13 @@ export default function Learning({ courses, learningModules, community }: { cour
...
         content={
           <>
-            <div className="text-base font-normal text-slate-700 pt-8 pb-7 md:w-182.5">{t("communities.overview.challenge.learning.title")}</div>
+            <div data-testid="learningId" className="text-base font-normal text-slate-700 pt-8 pb-7 md:w-182.5">
+              {t("communities.overview.challenge.learning.title")}
+            </div>
             <div className="md:grid grid-cols-2 gap-3">
               {courses?.map((course) => (
                 <LearningCard
+                  data-testid="cardId"
```
Comment (serapieTuyishime) [inline src/components/sections/challenges/Learning.tsx:33]:
```suggestion
                  data-testid="learning-card"
```

File: src/components/sections/challenges/Objectives.tsx [hunk 5]
```diff
@@ -23,8 +28,8 @@ export default function Objectives(): ReactElement {
   const expirationDate = challenge?.expiresAt && DateManager.format(challenge.expiresAt, "MMMM d, yyyy", "en");
   return (
     <Section title={`${t("communities.overview.challenge.objective.title")}`}>
-      <div className="space-y-5">
-        <ObjectiveList objectives={challenge?.objectives} />
+      <div data-testid="objectiveId" className="space-y-5">
+        <ObjectiveList objectives={challenge?.objectives || challenges?.objectives} />
```
Comment (serapieTuyishime) [inline src/components/sections/challenges/Objectives.tsx:32]: should this change be implemented in a testing PR?

File: src/components/sections/challenges/Objectives.tsx
Comment (serapieTuyishime) [inline src/components/sections/challenges/Objectives.tsx:1]: Was these changes reflected anywhere?

File: src/components/sections/challenges/_partials/HackathonPrize.tsx
Comment (serapieTuyishime) [inline src/components/sections/challenges/_partials/HackathonPrize.tsx:1]: Why was this change introduced

**Pull request closed without merging**